rbac_api.go

```diff
@@ -51,36 +50,66 @@ func (e *Enforcer) HasRoleForUser(name string, role string) (bool, error) {
 
 // AddRoleForUser adds a role for a user.
 // Returns false if the user already has the role (aka not affected).
-func (e *Enforcer) AddRoleForUser(user string, role string) (bool, error) {
-	return e.AddGroupingPolicy(user, role)
+func (e *Enforcer) AddRoleForUser(user string, role string, domain ...string) (bool, error) {
+	if len(domain) == 0 {
+		return e.AddGroupingPolicy(user, role)
+	} else if len(domain) > 1 {
+		return false, errors.ERR_DOMAIN_PARAMETER
+	}
+	return e.AddGroupingPolicy(user, role, domain[0])
 }
 
 // AddRolesForUser adds roles for a user.
 // Returns false if the user already has the roles (aka not affected).
-func (e *Enforcer) AddRolesForUser(user string, roles []string) (bool, error) {
+func (e *Enforcer) AddRolesForUser(user string, roles []string, domain ...string) (bool, error) {
 	f := false
-	for _, r := range roles {
-		b, err := e.AddGroupingPolicy(user, r)
-		if err != nil {
-			return false, err
+	if len(domain) == 0 {
+		for _, r := range roles {
```

Comment:
I believe we can use `batch` operation `AddGroupingPolicies`

Reply:
this is not the point of this PR.

Reply:
I can fix it in the next pr